staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go

```diff
@@ -26,9 +26,9 @@ import (
 // SetCRDCondition sets the status condition.  It either overwrites the existing one or
 // creates a new one
 func SetCRDCondition(crd *CustomResourceDefinition, newCondition CustomResourceDefinitionCondition) {
+	newCondition.LastTransitionTime = metav1.NewTime(time.Now())
 	existingCondition := FindCRDCondition(crd, newCondition.Type)
 	if existingCondition == nil {
-		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
 		crd.Status.Conditions = append(crd.Status.Conditions, newCondition)
 		return
 	}
```

Comment:
if we have an existing condition with a zero LastTransitionTime, we also need to update:
```go
if existingCondition.Status != newCondition.Status || existingCondition.LastTransitionTime.IsZero() {
```

Reply:
done